### PR TITLE
Support of MongoDB: Unified Topology Design

### PR DESCRIPTION
**Pull request by viktorzavadil**
I have added support of `useUnifiedTopology` option for MongoDB connection.

Feature requirements:
- you must provide own instance of MongoDB, see https://github.com/agenda/agenda#mongodbinstance
  ```js
  const client = new MongoClient(mongoCfg, {
    useNewUrlParser: true,
    useUnifiedTopology: true
  });
  
  await client.connect();
  
  const agenda = new Agenda({ mongo: client.db() });
  ```

Test of this feature is included.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -63,7 +63,7 @@
     "expect.js": "0.3.1",
     "jsdoc": "3.6.3",
     "jsdoc-template": "https://github.com/braintree/jsdoc-template",
-    "mocha": "6.2.2",
+    "mocha": "7.0.1",
```
Comment (simison) [inline package.json]: Why the Mocha update in this PR?

Comment (viktorzavadil) [reply package.json]: Ou, it seems `renovate-bot` update some dependencies. Maybe I used a bad branch name and that is the reason. What are the conventions?

**Comment by viktorzavadil** [comment]
@simison Should I fix the linter issues, downgrade mocha to 6.2.2, rename the branch and create new PR?

File: test/connection-use-unified-topology.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
+/* globals describe, it */
+'use strict';
+/* eslint object-curly-spacing: "off" */
+/* eslint space-before-function-paren: "off" */
+/* eslint prefer-arrow-callback: "warn"  */
```
[the remark this answers is not in the document]
Comment (viktorzavadil) [reply test/connection-use-unified-topology.js]: Yeah, it works.

I have different IDE settings and .editorconfig for this project is not enough for overriding all rules.

[the remark this answers is not in the document]
Comment (simison) [reply test/connection-use-unified-topology.js]: I can be useful to have Eslint integration installed in your editor directly, it should pick the right config in this folder I think. :-)

**Comment by simison** [comment]
You can rebase your branch against `origin/master` and force push to it, that should get things sortet without needing to create new PR.

**Comment by viktorzavadil** [comment]
@simison I have made a classic push into the old branch without rebasing and force push. What is the difference for you? If this is a problem I can try it but it seems like overkill :)

I fixed:
- mocha version,
- switched off linter rules.

**Comment by simison** [comment]
@viktorzavadil looks better! Can you please also take out all `yarn.lock` changes in this PR? Thanks!

**Comment by viktorzavadil** [comment]
@simison I forgot to change yarn changes after `renovate-bot` attack :smiley: I cherry-picked previous yarn version and I hope it will be good. I don't use yarn so I am not able to test it.

**Comment by simison** [comment]
It's fine, Yarn is run in CI and tested there. 👍

**Comment by MichielDeMey** [comment]
PR looks much cleaner now. 👍

**Pull request merged**